File: DSAIN90Days/Src/arrays/kadensAlgo/MaximumSubarray.java
package arrays.kadensAlgo;

public class MaximumSubarray {
	 public int maxSubArray(int[] nums) {
	  int max = nums[0];
      int currSum = max;
      for(int i = 1 ; i < nums.length; i++){
          currSum = Math.max(nums[i], currSum + nums[i]);
          max = Math.max(max ,currSum);
      }
      return max;
	 }
}

File: DSA_questions/Src/arrays/kadensAlgo/MaximumSubarray.py
import math
class Solution:
    def maxSubArray(self, nums):
        cur_max, max_till_now = 0, -math.inf
        for c in nums:
            cur_max = max(c, cur_max + c)
            max_till_now = max(max_till_now, cur_max)
        return max_till_now
    
if __name__ == "__main__":
    sol = Solution
    print(sol.maxSubArray(sol, [-2,1,-3,4,-1,2,1,-5,4]))